### PR TITLE
Armour Plating Degredation System

### PR DESCRIPTION
**Pull request by X0-11**
Adds a system that utilises an "armour thickness" variable to increase the effectiveness of armour, with the current armour variable-list acting as a lower bound for armour.

### CODE REVIEW

File: code/modules/halo/clothing/marine.dm [hunk 1]
```diff
@@ -42,6 +42,7 @@
 	icon_override = MARINE_OVERRIDE
 	icon_state = "M52B Body Armor"
 	blood_overlay_type = "armor"
+	body_parts_covered = ARMS|UPPER_TORSO|LOWER_TORSO
 	armor = list(melee = 50, bullet = 45, laser = 4, energy = 4, bomb = 60, bio = 0, rad = 0)
```
Comment (bloxgate) [inline code/modules/halo/clothing/marine.dm]: Uh... which of these protects from burns again?

**Review by bloxgate — APPROVED** [review]
Yeah. Looks good to me. Great work on getting this out fast.

File: code/modules/halo/clothing/marine.dm (hunk 1 above)
Comment (X0-11) [reply code/modules/halo/clothing/marine.dm]: Depends on the damage type of the projectile. It's probably energy.

Comment (bloxgate) [reply code/modules/halo/clothing/marine.dm]: Oh wow. No wonder marines were getting destroyed. 4 armor

Comment (X0-11) [reply code/modules/halo/clothing/marine.dm]: Yeeep. One second, I'll raise it. Suggested value?

Comment (bloxgate) [reply code/modules/halo/clothing/marine.dm]: Should be less than that of MJOLNIR for sure. Maybe 1/5 of that?

Comment (X0-11) [reply code/modules/halo/clothing/marine.dm]: MJOLNIR has it at 50, I think these values are balanced in the favor of the covenant.

Comment (bloxgate) [reply code/modules/halo/clothing/marine.dm]: Eh. I guess this is fine with the new mechanics then. Only testing will tell though.

**Review by BDpuffy420 — APPROVED** [review]

**Pull request merged**